cluster.tf

```diff
@@ -31,6 +31,11 @@ resource "aws_eks_cluster" "this" {
     aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy,
     aws_cloudwatch_log_group.this
   ]
+  provisioner "local-exec" {
```

Comment:
Any chance to avoid `local-exec` ?  The http provider doesn't work ? We wanted to be agnostic from the OS. Using the kubernetes provider was a huge improvement in that way and would like to avoid going back to local-exec with curl if possible.

Reply:
Ah ... missed that part of your comment; sorry! Hmm... I'd worry about three things: (1) That the http provider would sometimes timeout, and it doesn't provide any control over timeouts, (2) when not combined with the cluster resource itself, terraform could schedule other resources that depend on the cluster before its ready, and outside the module, its a bit opaque to know what to do (though it could be documented). (3) Doc says they verify chain of trust for https, and certificate is likely to be unchained.

I guess (2) could be possibly solved by having (say) cluster_id depend on an expression involving some output from http resource. (3) just testing will confirm/deny :)....

vs (1) -- I'm a terraform newbie ... do you know anything about http data source timeout behavior? Would seem to be a risk to me, but I'll go ahead and try it if you guys want to go that way.

Reply:
Most general would be to have variable for command be a template required to embed (say) ${URL}. I guess the shell varies by platform, but I guess this type of variable substitution should be broadly compatible? I'm not sure if the "until" loop is available "everywhere"? In any case, AFAIK the options are to code up a custom provider (or patch the http provider, to accept options for waiting, and for insecure https), or to try to make local-exec be as cross-platform as possible.

Reply:
> hen perhaps changing the kubernetes provider to wait until endpoint is up would also be an option.

https://github.com/terraform-providers/terraform-provider-kubernetes/issues/96

Reply:
I'm not sure what we should do now but open to opinions. Either merge with a new local-exec or think of something else 🙂

Reply:
I would think easiest way to fix something without upstream changes would be to use local-exec, but make it configurable for cross-platform support. Do you have an idea how you'd like to configure it? Do you want me to propose something? (Adding variables for configuration -- what program to use for "curl", for instance, or possibly whether to install curl appropriate for platform.)?

Reply:
From what I'm reading from stackoverflow, it sounds like since powershell 3.O, `wget` and `curl` are aliases for `Invoke-WebRequest` http://support.moonpoint.com/os/windows/PowerShell/wget-curl.php. It means, we can probably use `curl` in both OS, but i'm not sure for arguments.

Any windows guy around ?

Reply:
I think we forget about Windows. It's proved to be too much effort to support both platforms natively in the past. Windows users can use Docker I think.